src/Microsoft.PowerShell.SHiPS/Node/ContentWriter.cs

```diff
@@ -0,0 +1,133 @@
+﻿using System;
+using System.Collections;
+using System.IO;
+using System.Management.Automation.Provider;
+using CodeOwls.PowerShell.Provider.PathNodeProcessors;
+
+namespace Microsoft.PowerShell.SHiPS
+{
+    /// <summary>
+    /// The content stream class for the SHiPS provider. It implements the IContentWriter interfaces.
+    /// It is used in Set-Content cmdlet.
+    /// </summary>
+    internal class ContentWriter : IContentWriter
+    {
+        private MemoryStream _stream;
+        private StreamWriter _writer;
+        private readonly SHiPSBase _node;
+        private readonly SHiPSDrive _drive;
+        private readonly IProviderContext _context;
+
+        /// <summary>
+        /// Constructor for the content stream.
+        /// </summary>
+        /// <param name="context">A provider context.</param>
+        /// <param name="drive">SHiPS based provider drive.</param>
+        /// <param name="node">An object that is corresponding to the current node path.</param>
+        public ContentWriter(IProviderContext context, SHiPSDrive drive, SHiPSBase node)
+        {
+            _drive = drive;
+            _node = node;
+            _context = context;
+            _stream = new MemoryStream();
+            _writer = new StreamWriter(_stream);
+        }
+
+        /// <summary>
+        /// Moves the current stream position.
+        /// </summary>
+        ///
+        /// <param name="offset"> The offset from the origin to move the position to. </param>
+        /// <param name="origin"> The origin from which the offset is calculated. </param>
+        public void Seek(long offset, SeekOrigin origin)
+        {
+            _writer?.Flush();
+            _stream.Seek(offset, origin);
```

Comment:
not sure why writer gets ?. and stream does not

Reply:
fixed.thx.